test/unittests/evm_eip663_dupn_swapn_test.cpp

```diff
@@ -86,23 +80,20 @@ TEST_P(evm, dupn_full_stack)
 
     rev = EVMC_PRAGUE;
     auto full_stack_code = bytecode{};
-    for (uint64_t i = 1023; i >= 1; --i)
+    for (uint64_t i = 1022; i >= 1; --i)
```

Comment:
Why?

Reply:
oh yeah, this one. If we push 1023 times here, then the overall stack height goes up to 1024, which causes `max_stack_height_above_limit`, and we indeed have `t.max_stack_height > MAX_STACK_HEIGHT` and `constexpr auto MAX_STACK_HEIGHT = 0x03FF`, which is up to spec btw (megaspec).

The previous stack overflow check was at 1024.

A bit strange but correct I think? Or is there a problem with the spec?

Reply:
Make sense: in EOF the stack limit in a function is 1023.